tower_cli/api.py

```diff
@@ -262,9 +266,27 @@ def test_mode(self):
                                      password='This is the best wine.',
                                      verbose=False, format='json'):
             adapters = copy.copy(self.adapters)
-            faux_adapter = FauxAdapter(
-                url_pattern=self.prefix.rstrip('/') + '%s',
-            )
+
+            class VersionedFauxAdapter(FauxAdapter):
+                # An adapter that doesn't track version detection requests
+                # (those sent to /api/
+                def send(self, request, stream=False, **kwargs):
+                    ret = super(VersionedFauxAdapter, self).send(request, stream, **kwargs)
+                    if request.url.endswith('/api/'):
+                        self.requests.pop(-1)
+                    return ret
+
+                def clear(self):
+                    super(VersionedFauxAdapter, self).clear()
+                    self._url_pattern = '%s'
+                    self.register_json('https://%s/api/' % settings.host, {
```

Comment:
This registers - by default - a fake response for `/api/` so that version requests within tests succeed.  For now, I've left this hard-coded as `/api/v1/`, because that's what all of our tests expect.  In the future, as the stable version of the Tower API is incremented, we'll want to come in and change this value (and the `/api/v1/` strings hard-coded everywhere in our tests).